agroal-narayana/src/main/java/io/agroal/narayana/NarayanaTransactionIntegration.java

```diff
@@ -82,17 +83,21 @@ public NarayanaTransactionIntegration(TransactionManager transactionManager, Tra
 
     @Override
     public TransactionAware getTransactionAware() throws SQLException {
-        if ( getTransactionPhase() == TRANSACTION_ACTIVE ) {
+        TransactionPhase phase = getTransactionPhase();
+        if ( phase == TRANSACTION_ACTIVE ) {
             return (TransactionAware) transactionSynchronizationRegistry.getResource( key );
+        } else if ( phase == TRANSACTION_ROLLED_BACK ) {
```

Comment:
What about STATUS_ROLLING_BACK?

Reply:
`STATUS_ROLLING_BACK` is a state where the transaction is considered active, so the TSR is available